src/modules/products/services/products.service.ts

```diff
@@ -3,50 +3,66 @@ import { ProductBaseEntity } from '@entities/product-base.entity';
 import { ProductCategoryEntity } from '@entities/product-category.entity';
 import { ProductVariantEntity } from '@entities/product-variant.entity';
 import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { CloudinaryService } from '../../../modules/cloudinary/services/cloudinary.service';
 import { DataSource } from 'typeorm';
 
 @Injectable()
 export class ProductsService {
     constructor(
         private dataSource: DataSource,
+        private cloudinary: CloudinaryService
     ) { }
 
-    async createProduct(createProductDto: CreateProductDto) {
+    async createProduct(createProductDto: CreateProductDto, images: Array<Express.Multer.File>) {
         const queryRunner = this.dataSource.createQueryRunner();
         let productCategoryId = createProductDto.productCategoryId;
         let productBaseId = createProductDto.productBaseId;
         let savedProductCategory = null;
         let savedProductBase = null;
+        let productVariantImagesIds = [];
 
         await queryRunner.connect();
         await queryRunner.startTransaction();
         try {
             if (!productBaseId) {
                 if (!productCategoryId) {
-                    const newProductCategory = await queryRunner.manager.create(ProductCategoryEntity, createProductDto.productCategory);
+                    const newProductCategory = await queryRunner.manager.create(ProductCategoryEntity, { name: createProductDto.productCategoryName });
                     savedProductCategory = await queryRunner.manager.save(newProductCategory) as ProductCategoryEntity;
                 } else {
                     savedProductCategory = await queryRunner.manager.findOneByOrFail(ProductCategoryEntity, { id: productCategoryId }) as ProductCategoryEntity;
                 }
 
                 const newBaseProduct = await queryRunner.manager.create(ProductBaseEntity, {
-                    ...createProductDto.productBase,
+                    ...{ name: createProductDto.productBaseName },
```

Comment:
We don't need spread `...` operator here anymore. Can be simply `name: createProductDto.productBaseName`